prov/efa/src/rxr/rxr_cq.c

```diff
@@ -450,30 +451,61 @@ void rxr_cq_write_rx_completion(struct rxr_ep *ep,
 	efa_cntr_report_rx_completion(&ep->util_ep, rx_entry->cq_entry.flags);
 }
 
-void rxr_cq_handle_rx_completion(struct rxr_ep *ep,
-				 struct rxr_rx_entry *rx_entry)
+/**
+ * @brief complete an RX operation
+ *
+ * This function completes an RX operation. RX operation can be
+ * a receive, a write response, or a read response. This function
+ * is called when all data has been transmitted.
+ *
+ * To complete a RX operation, this function does 3 things:
+ *
+ * 1. If necessary, write completion to application. (Not all
+ *    completed RX action will cause a completion to be written).
+ *
+ * 2. When asked, send a ctrl packet back to the peer.
+ *    Caller of this function needs to decide whether such a ctrl packet
+ *    is needed. Usually it is because of the two scenarios:
+ *
+ *    a. longread message/write protocol was used, which requires the receiver/
+ *       responder send an EOR back to the sender/requester.
+ *
+ *    b. dc eager/medium/long message/write protocol was used, which requires
+ *       the receiver/responder to send an RECEIPT back to sender/requester.
+ *
+ * 3. Ensure release of rx_entry.
+ *
+ * @param[in,out]	ep		endpoint
+ * @param[in,out]	rx_entry	rx entry that contains information of the RX operation
+ * @param[in]		post_ctrl	whether to post a ctrl packet back to sender/requester
+ * @param[in]		ctrl_type	ctrl packet type.
+ */
+void rxr_cq_complete_rx(struct rxr_ep *ep,
```

Comment:
Are we going to do the same refactor for `rxr_cq_handle_tx_completion` in the future?

Reply:
Yes.